webpack/assets/javascripts/react_app/components/AutoComplete/AutoComplete.js

```diff
@@ -235,6 +243,7 @@ AutoComplete.defaultProps = {
   placeholder: 'Filter ...',
   emptyLabel: null,
   url: null,
+  id: `autocomplete-${Date.now()}`,
```

Comment:
Do we really want to encourage developers to use it? imo `id` should be required string.

Reply:
I mostly agree with you,
though I think there are some cases that the consumer doesn't care for the ID -
they just want the autocomplete to work.
In our case, some autocomplete implementation that I would like to replace are not using an id and defining each one will be difficult, 
that's why I chose to use this mechanism to give more choice.

Reply:
Shouldn't it actually suppose to be easy to define an `id` based on the consumer context?

Reply:
Before I'll change it, any other thoughts? 
cc @amirfefer @tbrisker